climada/util/plot.py

```diff
@@ -290,18 +331,143 @@ def geo_im_from_array(array_sub, coord, var_name, title,
+        -> categories mapping is [[0, 2, 1, 2], [0, 2, 3, 3]]
+        
+    Same category: 1 and '1'
+    Different categories: 1 and 1.0
+    
+    This method wraps around util.geo_scatter_from_array and uses 
+    all its args and kwargs.
+    
+    Parameters
+    ----------
+    array_sub : np.array(1d or 2d) or list(np.array)
+        Each array (in a row or in  the list) are values at each point 
+        in corresponding geo_coord that are binned in one subplot.
+    geo_coord : 2d np.array or list(2d np.array) 
+        (lat, lon) for each point in a row. If one provided, the same grid 
+        is used for all subplots. Otherwise provide as many as subplots
+        in array_sub.
+    var_name : str or list(str)
+        label to be shown in the colorbar. If one
+        provided, the same is used for all subplots. Otherwise provide as
+        many as subplots in array_sub.
+    title : str or list(str)
+        subplot title. If one provided, the same is
+        used for all subplots. Otherwise provide as many as subplots in
+        array_sub.
+    cat_name : dict, optional
+        Categories name for the colorbar labels. 
+        Keys are all the unique values in array_cub, values are their labels.
+        The default is labels = unique values.
+    pop_name : bool, optional
+        add names of the populated places. The default is False.
+    buffer : float, optional
+        border to add to coordinates. The default is BUFFER = 1.0.
+    extend : str, optional
+        extend border colorbar with arrows. The default is 'neither'.
+        Possible values : [ 'neither' | 'both' | 'min' | 'max' ]
+    proj : ccrs 
+        coordinate reference system used in coordinates.
+        The default is ccrs.PlateCarree()
+    kwargs : optional 
+        arguments for hexbin matplotlib function
+        
+    Returns
+    -------
+    cartopy.mpl.geoaxes.GeoAxesSubplot
+        
+    """
+    
+    # convert sorted categories to numeric array [0, 1, ...]
+    array_sub = np.array(array_sub)
+    array_sub_unique, array_sub_cat = np.unique(array_sub, return_inverse=True) #flattens array
+    array_sub_cat = array_sub_cat.reshape(array_sub.shape)
+    array_sub_n = array_sub_unique.size
+    
+    
+    #default qualitative cmap        
+    cmap_name = 'Dark2'
+    cmap = mpl.colors.ListedColormap(
+        plt.get_cmap(cmap_name).colors[:array_sub_n]
+        ) 
+
+    #optional user defined color map (can be continuous)
+    if 'cmap' in kwargs:
+        cmap = kwargs['cmap']
+        if isinstance(cmap, str):
+            cmap_name = cmap
+            cmap = mpl.cm.get_cmap(cmap)
+        else:
+            cmap_name = 'defined by the user'
+    
+    if array_sub_n > cmap.N:
+        LOGGER.warning("More than %d categories cannot be plotted accurately "
+                       "using the colormap %s. Please specify "
+                       "another qualitative colormap using the `cmap` "
```

Comment:
"another" != "a different"
```suggestion
                       "a different qualitative colormap using the `cmap` "
```